src/actions/guides/database/migrations.cr

```diff
@@ -0,0 +1,231 @@
+
+    > When you generate a new model, a migration file is created for you.
+
+    ### Anatomy
+
+    A new migration file will always start with a timestamp. This is to keep the order in which they are ran. You may have a migration that adds a column, and then a year later, another one that removes that column. You want to run these in the correct order.
+
+    In that file, you'll find two methods `migrate`, and `rollback`.
+
+    * migrate - Use this method to create your migration code.
+    * rollback - This method should un-do everything the `migrate` method does.
+
+    > The `rollback` method should do the opposite of `migrate` in reverse order. (e.g. If `migrate` adds a column, then adds an index for that column, `rollback` should remove the index first, then remove the column.)
+
+    ### DB Tasks
+
+    Lucky comes with several tasks you can run for handling your migrations.
+
+    * `db.migrate` - Run all pending migrations.
+    * `db.migrate.one` - Run only the next pending migration.
+    * `db.redo` - Rollback to the previous migration, then re-run the last migration again.
+    * `db.rollback` - Undo the last migration ran.
+    * `db.rollback_all` - Undo all of the migrations back to the beginning.
+
+    ## Create table
+
+    Use the `create` method for creating a table. You will place all of the table definitions inside of the `create` block. By default, Lucky will generate 3 columns for you with this: `id : Int32`, `created_at : Time`, and `updated_at : Time`.
+
+    ```crystal
+    def migrate
+      create :users do
+        # add column definitions here
+      end
+    end
+    ```
+
+    Optionally, if you need your `id` column to be of type `UUID`, you can specify the `primary_key_type: :uuid` option.
+
+    ```crystal
+    def migrate
+      create :users, primary_key_type: :uuid do
+        add email : String
+      end
+    end
+    ```
+
+    > Be sure to update your model with `primary_key_type: :uuid` as well.
+
+    ## Drop table
+
+    To drop a table, use the `drop` method.
+
+    ```crystal
+    def migrate
+      drop :users
+    end
+    ```
+
+    > If your `migrate` method runs `create`, then the `rollback` method should run `drop`.
+
+    ## Alter table
+
+    If your table exists, but you need to make changes to it, use the `alter` method. All of your changes will go in the `alter` block.
+
+    ```crystal
+    def migrate
+      alter :users do
+        remove :old_field
+      end
+    end
+    ```
+
+    ## Add column
+
+    To add a new column, you'll use the `add` method inside of a `[create](#create-table)` block.
+
+    ```crystal
+    create :users do
+      add email : String
+      add birthdate : Time
+      add login_count : Int32
+    end
+    ```
+
```

Comment:
I think a section on fill_existing_with would be helpful. Basically if you have an alter table with a required column it requires you to use fill_existing with. That can be a value, or :do_nothing

The idea is that when you add a required column to an existing table with rows it will fail. So we make it easy to fill existing columns with a default before applying the non null constraint. 

You can use :do_nothing though if the rows are empty or if you plan to handle data migration later